# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1464594163 18000
#      Mon May 30 02:42:43 2016 -0500
# Node ID b2f1f4cf13892e8ee7fc5995593d3dbbb4348002
# Parent  a4c744194bbb825e543059953e980072c7893616
Correct linestyle property in toolkit for old versions of gnuplot (bug #48033)

* __gnuplot_draw_axes__.m: Set style to 'lines', regardless of linetype lt
  being empty.
  (do_linestyle_c): Ditto.  Also change 'line style' to 'style line' of
  gnuplot linestyle configuration command.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -913,21 +913,17 @@ function __gnuplot_draw_axes__ (h, plot_
                ccol = 255*ccol*[0x1; 0x100; 0x10000];
              endif
            else
              colorspec = sprintf ("lc rgb \"#%02x%02x%02x\"",
                                   round (255*color));
            endif
 
            sidx = 1;
-           if (isempty (lt))
-             style = "";
-           else
-             style = "lines";
-           endif
+           style = "lines";
            tmpwith = {};
 
            facesame = true;
            if (! isequal (pt, pt2) && isfield (obj, "markerfacecolor")
                && ! strcmp (obj.markerfacecolor, "none"))
              if (strcmp (obj.markerfacecolor, "auto")
                  || ! isnumeric (obj.markerfacecolor)
                  || (isnumeric (obj.markerfacecolor)
@@ -1812,17 +1808,17 @@ function [style, ltidx] = do_linestyle_c
                                                 plot_stream, errbars = "")
   idx = idx + 8;
   style = {};
   ltidx = [];
 
   if (__gnuplot_has_feature__ ("linetype"))
     scommand = "linetype";
   else
-    scommand = "line style";
+    scommand = "style line";
   endif
   fprintf (plot_stream, "set %s %d default;\n", scommand, idx);
   fprintf (plot_stream, "set %s %d", scommand, idx);
 
   found_style = false;
   if (isnumeric (linecolor))
     color = linecolor;
     fprintf (plot_stream, " linecolor rgb \"#%02x%02x%02x\"",
@@ -1847,21 +1843,17 @@ function [style, ltidx] = do_linestyle_c
   [pt, pt2, obj] = gnuplot_pointtype (obj);
 
   if (! isempty (pt))
     found_style = true;
   endif
 
   sidx = 1;
   if (isempty (errbars))
-    if (isempty (lt))
-      style{sidx} = "";
-    else
-      style{sidx} = "lines";
-    endif
+    style{sidx} = "lines";
     ltidx(sidx) = idx;
 
     facesame = true;
     if (! isequal (pt, pt2) && isfield (obj, "markerfacecolor")
         && ! strcmp (obj.markerfacecolor, "none"))
       if (strcmp (obj.markerfacecolor, "auto")
           || ! isnumeric (obj.markerfacecolor)
           || (isnumeric (obj.markerfacecolor)
